src/gsy_e/gsy_e_core/rq_job_handler.py

```diff
@@ -148,3 +76,95 @@ def launch_simulation_from_rq_job(scenario: Dict,
+            ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS = True
+            # Disable fully automatic trading mode for the template strategies in favor of
+            # UI manual and auto modes.
+            ConstSettings.ForwardMarketSettings.FULLY_AUTO_TRADING = False
+
+    gsy_e.constants.SEND_EVENTS_RESPONSES_TO_SDK_VIA_RQ = True
+
+    spot_market_type = settings.get("spot_market_type")
+    bid_offer_match_algo = settings.get("bid_offer_match_algo")
+
+    if spot_market_type:
+        ConstSettings.MASettings.MARKET_TYPE = spot_market_type
+    if bid_offer_match_algo:
+        ConstSettings.MASettings.BID_OFFER_MATCH_TYPE = bid_offer_match_algo
+
+    ConstSettings.SettlementMarketSettings.RELATIVE_STD_FROM_FORECAST_FLOAT = (
+        settings.get(
+            "relative_std_from_forecast_percent",
+            ConstSettings.SettlementMarketSettings.RELATIVE_STD_FROM_FORECAST_FLOAT
+        ))
+
+    ConstSettings.SettlementMarketSettings.ENABLE_SETTLEMENT_MARKETS = settings.get(
+        "settlement_market_enabled",
+        ConstSettings.SettlementMarketSettings.ENABLE_SETTLEMENT_MARKETS
+    )
+    gsy_e.constants.CONNECT_TO_PROFILES_DB = connect_to_profiles_db
+
+
+def _create_config_settings_object(
+        scenario: Dict, settings: Dict, aggregator_device_mapping: Dict) -> SimulationConfig:
+    config_settings = {
+        "start_date":
+            instance(datetime.combine(settings.get("start_date"), datetime.min.time()))
+            if "start_date" in settings else GlobalConfig.start_date,
+        "sim_duration":
+            duration(days=settings["duration"].days)
+            if "duration" in settings else GlobalConfig.sim_duration,
+        "slot_length":
+            duration(seconds=settings["slot_length"].seconds)
+            if "slot_length" in settings else GlobalConfig.slot_length,
+        "tick_length":
+            duration(seconds=settings["tick_length"].seconds)
+            if "tick_length" in settings else GlobalConfig.tick_length,
+        "market_maker_rate":
+            settings.get("market_maker_rate",
+                         ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE),
+        "cloud_coverage": settings.get("cloud_coverage", GlobalConfig.cloud_coverage),
+        "pv_user_profile": settings.get("pv_user_profile", None),
+        "capacity_kW": settings.get("capacity_kW",
+                                    ConstSettings.PVSettings.DEFAULT_CAPACITY_KW),
+        "grid_fee_type": settings.get("grid_fee_type", GlobalConfig.grid_fee_type),
+        "external_connection_enabled": settings.get("external_connection_enabled", False),
+        "aggregator_device_mapping": aggregator_device_mapping
+    }
+
+    if GlobalConfig.IS_CANARY_NETWORK:
+        config_settings["start_date"] = (
+            instance((datetime.combine(date.today(), datetime.min.time()))))
+        if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS.value:
```

Comment:
I think it is time to create a utility function for this. If you agree, I can do it in the live events story.

Reply:
Agreed and thanks in advance!